### PR TITLE
Allow SunPy to be installed without astropy

### LINKED ISSUE #1516: Remove the need to have astropy installed before installing SunPy
Currently you can not have a clean python environment and do a `pip install sunpy` you have to have astropy + numpy installed first.

### PR DESCRIPTION
**Pull request by Cadair**
closes #1516

### CODE REVIEW

**Comment by ayshih** [comment]
Yay, I am totally on board with removing `setup_requires=['numpy>1.7.1']` (see #1396)!

**Comment by Cadair** [comment]
@ayshih yeah that doesn't work. Removing it from the setup requires means it will not install if it is not installed already.

**Comment by ayshih** [comment]
Rats.  But, it's in `install_requires`, so presumably the issue is that we need NumPy during setup?

**Comment by Cadair** [comment]
@ayshih yeah astropy_helpers needs it during the setup process.

**Comment by ayshih** [comment]
Shouldn't you remove "or numpy" from the issue name?

**Comment by ehsteve** [comment]
We need someone to actually test this.

**Comment by Cadair** [comment]
I have tested it :p

**Comment by ayshih** [comment]
Tests to my satisfaction :+1:

File: setup.py [hunk 1]
```diff
@@ -64,9 +63,14 @@
...
-# Overwrite the Astropy Testing framework
-cmdclassd['test'] = type('SunPyTest', (SunPyTest,),
-                         {'package_name': 'sunpy'})
+try:
+    from sunpy.tests.setup_command import SunPyTest
+    # Overwrite the Astropy Testing framework
+    cmdclassd['test'] = type('SunPyTest', (SunPyTest,),
+                            {'package_name': 'sunpy'})
+except Exception:
+    # Get everything, if it dosen't work, we still want SunPy to install.
```
Comment (ayshih) [inline setup.py]: I don't understand the start of this comment.  There's also a spelling error.

**Pull request merged**